.github/workflows/main.yaml

```diff
@@ -11,7 +11,7 @@ on:
 jobs:
   build:
     name: Build and push Docker image
-    runs-on: ubuntu-20.04
+    runs-on: [self-hosted,linux,x64]
```

Comment:
This repository is public github doesn't charge for building minutes. No need to run it on self-hosted.